### PR TITLE
BaSP-TG-86: Timesheets redux - Create, update and delete

### PR DESCRIPTION
**Pull request by Ansitox**
- Create constants to create, update and delete timesheets
- Create actions to create, update and delete timesheets
- Create thunk to create, update and delete timesheets
- Complete timesheets reducer to create, update and delete
- Connect list component with store to get and delete timesheets
- Conectect form with store to update and create timesheets

For QAs:
See how to test on Trello:
[Ticket 86](https://trello.com/c/8YllfrV9)

### CODE REVIEW

File: src/redux/timesheets/actions.js [hunk 1]
```diff
@@ -2,7 +2,19 @@ import {
   GET_TIMESHEETS_PENDING,
   GET_TIMESHEETS_SUCCESS,
   GET_TIMESHEETS_ERROR,
-  TOGGLE_MESSAGE_MODAL
+  DELETE_TIMESHEETS_PENDING,
+  DELETE_TIMESHEETS_SUCCES,
```
Comment (renzopimentel) [inline src/redux/timesheets/actions.js:6]: Try to keep the same name for each request, in this case you should put SUCCESS like in GET request

File: src/redux/timesheets/actions.js [hunk 2]
```diff
@@ -2,7 +2,19 @@ import {
...
+  DELETE_TIMESHEETS_ERROR,
+  POST_TIMESHEETS_PENDING,
+  POST_TIMESHEETS_SUCCES,
```
Comment (renzopimentel) [inline src/redux/timesheets/actions.js:9]: Same as above

File: src/redux/timesheets/actions.js [hunk 3]
```diff
@@ -2,7 +2,19 @@ import {
...
+  POST_TIMESHEETS_ERROR,
+  PUT_TIMESHEETS_PENDING,
+  PUT_TIMESHEETS_SUCCES,
```
Comment (renzopimentel) [inline src/redux/timesheets/actions.js:12]: Same as above

File: src/redux/timesheets/actions.js [hunk 4]
```diff
@@ -25,8 +37,89 @@ export const getTimesheetsError = (error) => {
...
+export const deleteTimesheetsPending = () => {
   return {
-    type: TOGGLE_MESSAGE_MODAL
+    type: DELETE_TIMESHEETS_PENDING
+  };
+};
+
+export const deleteTimesheetsSuccess = (id) => {
+  return {
+    type: DELETE_TIMESHEETS_SUCCES,
```
Comment (renzopimentel) [inline src/redux/timesheets/actions.js:48]: And change in the types too

File: src/redux/timesheets/constants.js [hunk 5]
```diff
@@ -1,4 +1,20 @@
 export const GET_TIMESHEETS_PENDING = 'GET_TIMESHEETS_PENDING';
 export const GET_TIMESHEETS_SUCCESS = 'GET_TIMESHEETS_SUCCESS';
 export const GET_TIMESHEETS_ERROR = 'GET_TIMESHEETS_ERROR';
-export const TOGGLE_MESSAGE_MODAL = 'TOGGLE_MESSAGE_MODAL';
+
+export const DELETE_TIMESHEETS_PENDING = 'DELETE_TIMESHEETS_PENDING';
+export const DELETE_TIMESHEETS_SUCCES = 'DELETE_TIMESHEETS_SUCCES';
```
Comment (renzopimentel) [inline src/redux/timesheets/constants.js:6]: Same as described in actions

File: src/redux/timesheets/reducer.js [hunk 6]
```diff
@@ -2,15 +2,28 @@ import {
   GET_TIMESHEETS_PENDING,
   GET_TIMESHEETS_SUCCESS,
   GET_TIMESHEETS_ERROR,
-  TOGGLE_MESSAGE_MODAL
+  DELETE_TIMESHEETS_PENDING,
+  DELETE_TIMESHEETS_SUCCES,
```
Comment (renzopimentel) [inline src/redux/timesheets/reducer.js:6]: Same as described in actions

File: src/redux/timesheets/thunks.js [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+    }
+  };
+};
+
+export const deleteTimeSheet = (id) => {
+  return async (dispatch) => {
+    dispatch(deleteTimesheetsPending());
+    try {
+      const response = await fetch(`${process.env.REACT_APP_API_URL}/timesheets/${id}`, {
+        method: 'DELETE'
```
Comment (renzopimentel) [inline src/redux/timesheets/thunks.js:66]: Its important to declare the content type of the header as 'application/json' when sending or receiving JSON from the server

File: src/redux/timesheets/thunks.js [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+          date: input.date,
+          hours: input.hours,
+          task: input.task,
+          employee: input.employee,
+          project: input.project
+        })
+      });
+      if (response.status == 200) {
+        const data = await response.json();
+        console.log(data);
```
Comment (CristianBucci) [inline src/redux/timesheets/thunks.js:101]: Remove this console.log!

File: src/redux/timesheets/actions.js (hunk 1 above)
Comment (Ansitox) [reply src/redux/timesheets/actions.js:6]: done

File: src/redux/timesheets/actions.js (hunk 2 above)
Comment (Ansitox) [reply src/redux/timesheets/actions.js:9]: done

File: src/redux/timesheets/constants.js (hunk 5 above)
Comment (Ansitox) [reply src/redux/timesheets/constants.js:6]: done

File: src/redux/timesheets/reducer.js (hunk 6 above)
Comment (Ansitox) [reply src/redux/timesheets/reducer.js:6]: done

File: src/redux/timesheets/thunks.js (hunk 7 above)
Comment (Ansitox) [reply src/redux/timesheets/thunks.js:66]: done

File: src/redux/timesheets/thunks.js [hunk 9]
```diff
@@ -0,0 +1,112 @@
...
+        dispatch(putTimesheetsSuccess(data.data, data.message));
+      } else {
+        const data = await response.json();
+        dispatch(putTimesheetsError(data.data));
+      }
+    } catch (error) {
+      console.log(error);
```
[the remark this answers is not in the document]
Comment (Ansitox) [reply src/redux/timesheets/thunks.js:108]: done

File: src/redux/timesheets/thunks.js (hunk 8 above)
Comment (Ansitox) [reply src/redux/timesheets/thunks.js:101]: done

File: src/Components/TimeSheets/Form/index.js [hunk 10]
```diff
@@ -82,121 +66,63 @@ const Form = (props) => {
...
+  const onCancel = () => {
+    dispatch(confirmModalClose());
+  };
+
+  const redirect = () => {
+    props.history.push('/timesheets');
+  };
+
+  const modalFunction = () => {
+    modalContent.title.includes('SUCCESS') ? redirect() : null;
```
Comment (MaxiG-dev) [inline src/Components/TimeSheets/Form/index.js:95]: You can replace 
`modalContent.title.includes('SUCCESS') ? redirect() : null;`
with
`modalContent.title.includes('SUCCESS') && redirect();`

File: src/Components/TimeSheets/index.js [hunk 11]
```diff
@@ -1,38 +1,56 @@
...
 
-  useEffect(() => {
-    dispatch(getTimesheets());
-  }, []);
+  const redirect = () => {
+    props.history.push('/timesheets');
+  };
+
+  const modalFunction = () => {
+    modalContent.title.includes('SUCCESS') ? redirect() : null;
```
Comment (MaxiG-dev) [inline src/Components/TimeSheets/index.js:45]: You can replace 
`modalContent.title.includes('SUCCESS') ? redirect() : null;`
with
`modalContent.title.includes('SUCCESS') && redirect();`

File: src/Components/TimeSheets/Form/index.js (hunk 10 above)
Comment (Ansitox) [reply src/Components/TimeSheets/Form/index.js:95]: done

File: src/Components/TimeSheets/index.js (hunk 11 above)
Comment (Ansitox) [reply src/Components/TimeSheets/index.js:45]: done

File: src/Components/TimeSheets/Form/index.js [hunk 12]
```diff
@@ -1,14 +1,36 @@
...
 
 const Form = (props) => {
+  const dispatch = useDispatch();
+
+  const { modalContent, showModalMessage, showConfirmModal } = useSelector(
+    (state) => state.timesheets
+  );
+
+  const params = useParams();
+  const id = params.id ? params.id : '';
```
Comment (ignacioaristo) [inline src/Components/TimeSheets/Form/index.js:33]: we could use && operator

**Review by ignacioaristo — APPROVED** [review]

**Review by CristianBucci — APPROVED** [review]
Everything looks right!

**Review by julianpiaggio — APPROVED** [review]
🕵️‍♂️ QA FEEDBACK

We wanted to tell you that we consider that your pr complies
the requirements to be approved, and that it can be merged into the master branch 👍

**Pull request merged**